HOTFIX:  WCOSS2 build issue

Following implementation of RFC 13644 the WCOSS2 GDASApp build fails with `undefined reference` messages
```
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: /opt/cray/pe/lib64/libmpifort_intel.so.12: undefined reference to `PMPIX_Win_post_stream'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: /opt/cray/pe/lib64/libmpifort_intel.so.12: undefined reference to `PMPIX_Free_queue'
```

GDIT informed us that we need to add the following line to the GDASApp build script
```
export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/opt/cray/pe/mpich/8.1.19/ofi/intel/19.0/lib"
```

FYI, here is the RFC memo description of RFC 13644

> RFC 13644 - On WCOSS2, set default cray_pe/cray-mpich versions and remove
> 
> the python preload for ESMF. This will resolve conflicts between the default
> programming environment and the default craype environment modules. The
> cray-mpich setting resolves the need for the preload change in the ESMF module
> (that change - RFC 13432, implemented on November 13, 2024 - is being
> reverted). To be implemented on February 5 (backup system) and February 6
> (primary system) from 1500Z to 1600Z each day.

This issue is opened to document this problem and implement a fix

--- a/build.sh
+++ b/build.sh
@@ -89,6 +89,11 @@ esac
 
 CMAKE_OPTS+=" -DCLONE_JCSDADATA=$CLONE_JCSDADATA -DMACHINE=$BUILD_TARGET"
 
+# TODO: Remove LD_LIBRARY_PATH line as soon as permanent solution is available
+if [[ $BUILD_TARGET == 'wcoss2' ]]; then
+    export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/opt/cray/pe/mpich/8.1.19/ofi/intel/19.0/lib"
+fi
+
 BUILD_DIR=${BUILD_DIR:-$dir_root/build}
 if [[ $CLEAN_BUILD == 'YES' ]]; then
   [[ -d ${BUILD_DIR} ]] && rm -rf ${BUILD_DIR}

